packages/cdktf/lib/terraform-local.ts

```diff
@@ -16,27 +17,27 @@ export class TerraformLocal extends TerraformElement {
   }
 
   public get expression() {
-    return Token.asAny(this.interpolation());
+    return Token.asAny(ref(this.interpolation()));
   }
 
   public get asString(): string {
-    return Token.asString(this.interpolation());
+    return Token.asString(ref(this.interpolation()));
   }
 
   public get asNumber(): number {
-    return Token.asNumber(this.interpolation());
+    return Token.asNumber(ref(this.interpolation()));
   }
 
   public get asList(): string[] {
-    return Token.asList(this.interpolation());
+    return Token.asList(ref(this.interpolation()));
   }
 
   public get asBoolean(): boolean {
-    return Token.asString(this.interpolation()) as any as boolean;
+    return Token.asString(ref(this.interpolation())) as any as boolean;
```

Comment:
Not related to this PR, but this cast has never worked correctly and still won't after this change.

Reply:
I changed the boolean interfaces now to take boolean and IResolvable, so I will return the ref directly 👍